File: HardwareSpot/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Spatie\Sluggable\HasSlug;
use Spatie\Sluggable\SlugOptions;


class Product extends Model
{
    use HasFactory;
    use HasSlug;

    protected $fillable = ['name', 'description', 'price', 'descount', 'quantity', 'slug'];

    public function getSlugOptions() : SlugOptions {

        return SlugOptions::create()->generateSlugsFrom('name')->saveSlugsTo('slug');

    }

    public function categories() {
        return $this->belongsToMany(Category::class);
    }

    public function photos(){
        return $this->hasMany(ProductPhoto::class);
    }

    public function getCategories(){
        return response()->json(Category::with('product','photos')->get());

    }
    
    public function cart(){ 

        return $this->belongsToMany(Cart::class)->withPivot('amount');

    }
}

File: Mobile/app/src/main/java/br/app/pi4mobile/models/Product.kt
package br.app.pi4mobile.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Product(
    val id: Int,
    val name: String,
    val description: String,
    val price: String,
    val descount: String, //discount?
    val quantity: String,
    val slug: String,
    val photos: List<Photo>? = null

): Parcelable
